### PR TITLE
[RHPAM-1272] Automated Task reassignment is not working

### PR DESCRIPTION
**Pull request by elguardian**

### CODE REVIEW

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/DeadlineSchedulerHelper.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.kie.internal.task.api.model.Deadline;
+import org.kie.internal.task.api.model.Deadlines;
+import org.kie.internal.task.api.model.InternalTask;
+
+
+public class DeadlineSchedulerHelper {
+
+    public static void rescheduleDeadlineForTask(Task task, TaskContext ctx, Deadline deadline, DeadlineType type) {
+        // here is the problem I cannot access the definition during Execute deadlines command
+        // I don't see how to access which
```
Comment (mswiderski) [inline jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/DeadlineSchedulerHelper.java]: What definition you have in mind?

**Comment by elguardian** [comment]
The one creating the deadline. How shoud i know how to regerate the deadline. I know during add task how to create the deadlines (all of them) but in this case one specific deadline has bean expired. I mean by def the string input var like users:Tony Stark,Bobba Fet,Jabba Hutt|groups:]@[500ms]@notcompleted

**Comment by elguardian** [comment]
@mswiderski  do you mind to have a look ?

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/DeadlinesBaseTest.java [hunk 2]
```diff
@@ -461,4 +471,91 @@ public void testDelayedEmailNotificationOnDeadlineTaskCompletedMultipleTasks() t
...
+          Environment environment = EnvironmentFactory.newEnvironment();
+          Map<String, Object> inputVars = new HashMap<String, Object>();
+          inputVars.put("NotCompletedReassign", "[users:Tony Stark,Bobba Fet,Jabba Hutt|groups:]@[500ms]");
+          ((InternalTask) task).setDeadlines(HumanTaskHandlerHelper.setDeadlines(inputVars, Collections.emptyList(), environment));
+
+          taskService.addTask(task, inputVars);
+
+          long taskId = task.getId();
+
+          task = (InternalTask) taskService.getTaskById(taskId);
```
Comment (MarianMacik) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/DeadlinesBaseTest.java]: Is this particular result of this method call used anywhere?

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/DeadlinesBaseTest.java [hunk 3]
```diff
@@ -461,4 +471,91 @@ public void testDelayedEmailNotificationOnDeadlineTaskCompletedMultipleTasks() t
...
+          for(String owner : owners) {
+              CountDownTaskEventListener countDownListener = new CountDownTaskEventListener(1, true, false);
+              addCountDownListner(countDownListener);
+
+              taskService.claim(taskId, owner);
+              task = (InternalTask) taskService.getTaskById(taskId);
+              Assert.assertEquals(task.getTaskData().getActualOwner().getId(), owner);
+
+              taskService.start(taskId, owner);
+              task = (InternalTask) taskService.getTaskById(taskId);
```
Comment (MarianMacik) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/DeadlinesBaseTest.java]: This call is not used anywhere, so can be omitted I think.

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/DeadlinesBaseTest.java [hunk 4]
```diff
@@ -461,4 +471,91 @@ public void testDelayedEmailNotificationOnDeadlineTaskCompletedMultipleTasks() t
...
+
+          Map<String, Object> inputVars = new HashMap<String, Object>();
+          inputVars.put("NotStartedReassign", "[users:Tony Stark,Bobba Fet,Jabba Hutt|groups:]@[500ms]");
+          ((InternalTask) task).setDeadlines(HumanTaskHandlerHelper.setDeadlines(inputVars, Collections.emptyList(), environment));
+
+          taskService.addTask(task, inputVars);
+
+          long taskId = task.getId();
+
+          task = taskService.getTaskById(taskId);
```
Comment (MarianMacik) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/DeadlinesBaseTest.java]: This code is not needed as it seems. The result is not used anywhere.

**Comment by elguardian** [comment]
@MarianMacik, I think I didn't miss anything

**Comment by elguardian** [comment]
@MarianMacik just added the CountDownTaskEventListener as suggested by @mswiderski

**Comment by mswiderski** [comment]
@mareknovotny why this has label DO NOT MERGE?

**Comment by mareknovotny** [comment]
@mswiderski I think it wasn't properly agreed to include it, so if you are ok, we need to follow on https://issues.jboss.org/browse/RHPAM-1272 and do it ;)

**Comment by mswiderski** [comment]
@mareknovotny it was not agreed to be added to 6.x but as feature request it makes sense to be improved and this PR is targeting master, or am I missing something?

**Comment by mareknovotny** [comment]
@mswiderski you're right, but at time I added the label, this PR was still in review and discussed, I will remove label now ;).

**Comment by mareknovotny** [comment]
I don't like the text in PR title Don't merge so I change it to label as that is more clear for me to orient between them ;)

**Comment by mswiderski** [comment]
@mareknovotny full agree about the text and already started to use the label as well so thanks for that hint!

**Comment by mareknovotny** [comment]
so nothing holds us to merge it ;-)

**Pull request merged**